tools/auto-fuzz/fuzz_driver_generation_jvm.py

```diff
@@ -0,0 +1,292 @@
+                                               init_dict)
+            if result:
+                return result
+
+    return None
+
+
+def _handle_object_creation(classname, init_dict, possible_target):
+    """
+    Generate statement for Java object creation of the target class.
+    If constructor (<init>) does existed in the yaml file, we will
+    use it as reference, otherwise the default empty constructor
+    are used.
+    """
+    if init_dict and classname in init_dict.keys():
+        # Process arguments for constructor
+        try:
+            arg_list = []
+            func_elem = init_dict[classname]
+
+            if not func_elem['JavaMethodInfo']['classConcrete']:
+                func_elem = _search_concrete_subclass(classname, init_dict)
+            if not func_elem:
+                return "new " + classname.replace("$", ".") + "()"
+
+            classname = func_elem['functionSourceFile']
+            for argType in func_elem['argTypes']:
+                arg_list.append(
+                    _handle_argument(argType, init_dict, possible_target))
+            possible_target.exceptions_to_handle.update(
+                func_elem['JavaMethodInfo']['exceptions'])
+            possible_target.imports_to_add.update(_handle_import(func_elem))
+            return "new " + classname.replace(
+                "$", ".") + "(" + ",".join(arg_list) + ")"
+        except RecursionError:
+            # Fail to create constructor code with parameters, using default constructor
+            return "new " + classname.replace("$", ".") + "()"
+    else:
+        return "new " + classname.replace("$", ".") + "()"
+
+
+def _generate_heuristic_1(yaml_dict, possible_targets):
+    """Heuristic 1.
+    Creates a FuzzTarget for each method that satisfy all:
+        - public class method which are not abstract or found in JDK library
+        - have between 0-20 arguments
+        - do not have "test" in the function name
+    The fuzz target is simply one that calls into the target function with
+    a string seeded with fuzz data.
+
+    Will also add proper exception handling based on the exception list
+    provided by the frontend code.
+    """
+    HEURISTIC_NAME = "jvm-autofuzz-heuristics-1"
+    for func_elem in yaml_dict['All functions']['Elements']:
+        java_method_info = func_elem['JavaMethodInfo']
+
+        # Skip method which doese not match this heuristic
+        if not java_method_info['static']:
+            continue
+        if not java_method_info['public']:
+            continue
+        if not java_method_info['concrete']:
+            continue
+        if java_method_info['javaLibraryMethod']:
+            continue
+        if len(func_elem['argTypes']) > 20:
+            continue
+        if "test" in func_elem['functionName']:
+            continue
+
+        possible_target = FuzzTarget()
+
+        # Store target method name
+        func_name = func_elem['functionName'].split('].')[1].split('(')[0]
```

Comment:
Could we add a simple example  as a comment on why this string manipulation is necessary?